File: framework/lib/event_args.dart
part of uxml;

/**
 * Provides base class for event arguments.
 *
 * @author ferhat@ (Ferhat Buyukkokten)
 */
class EventArgs {
  EventArgs(Object eventSource) {
    source = eventSource;
    currentSource = eventSource;
    handled = false;
  }

  /** Used to expose source event definition to handlers */
  EventDef event;

  /** Set if event was handled during routing */
  bool handled;

  /** The object that originated this event. */
  Object source;

  /** The object that dispatched this event (in case of event bubbling). */
  Object currentSource;

  /**
   *  For platform level input events although an element marks the
   * event as handled, we need platform to perform default processing.
   * This flag is set by MouseEventArgs._passthrough();
   */
  bool _forceDefault = false;

  /**
   * Initializes arguments so the instance can be reused.
   */
  void reset() {
    handled = false;
    _forceDefault = false;
  }
}

class Route {
  int route;
  Route(this.route);

  static Route DIRECT;
  static Route BUBBLE;
  static Route TUNNEL;

  static void _initRoute() {
    DIRECT = new Route(0);
    BUBBLE = new Route(1);
    TUNNEL = new Route(2);
  }
}

typedef void EventDefHandler(UxmlElement element, EventArgs e);

File: plugin/com.hello.uxml.tools.ui/src/com/hello/uxml/tools/framework/events/EventArgs.java
package com.hello.uxml.tools.framework.events;

/**
 * Provides base class for event arguments.
 *
 * @author ferhat@(Ferhat Buyukkokten)
 */
public class EventArgs {

  /** Set if event was handled during routing */
  protected boolean handled;

  /** Used to expose source event definition to handlers */
  protected EventDefinition eventDef;

  /** The object that fired this event. */
  protected Object source;

  /**
   * Constructor.
   */
  public EventArgs() {
  }

  /**
   * Constructor.
   */
  public EventArgs(Object source, EventDefinition eventDef) {
    this.source = source;
    this.eventDef = eventDef;
  }

  /**
   * Sets or returns if an event was handled.
   */
  public boolean getHandled() {
    return handled;
  }

  public void setHandled(boolean value) {
    handled = value;
  }

  /**
   * Sets or returns event definition
   */
  public EventDefinition getEvent() {
    return eventDef;
  }

  public void setEvent(EventDefinition event) {
    eventDef = event;
  }

  /**
   * Gets or sets the source of the event.
   */
  public Object getSource() {
    return source;
  }

  public void setSource(Object value) {
    source = value;
  }
}
